pkg/controller/daemon/daemoncontroller.go

```diff
@@ -812,20 +813,42 @@ func (dsc *DaemonSetsController) syncNodes(ds *extensions.DaemonSet, podsToDelet
 
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v, creating %d", ds.Name, nodesNeedingDaemonPods, createDiff)
 	createWait := sync.WaitGroup{}
-	createWait.Add(createDiff)
 	template := util.CreatePodTemplate(ds.Spec.Template, ds.Spec.TemplateGeneration, hash)
-	for i := 0; i < createDiff; i++ {
-		go func(ix int) {
-			defer createWait.Done()
-			if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, &template, ds, newControllerRef(ds)); err != nil {
-				glog.V(2).Infof("Failed creation, decrementing expectations for set %q/%q", ds.Namespace, ds.Name)
+	// Batch the pod creates. Batch sizes start at 1 and double with each
+	// successful iteration in a kind of "slow start". This handles attempts
+	// to start large numbers of pods that would likely all fail with the
+	// same error. For example a project with a low quota that attempts to
+	// create a large number of pods will be prevented from spamming the
+	// API service with the pod create requests after one of its pods fails.
+	// Conveniently, this also prevents the event spam that those failures
+	// would generate.
+	for batchSize, pos := 1, 0; createDiff > pos; batchSize, pos = integer.IntMin(2*batchSize, createDiff-(pos+batchSize)), pos+batchSize {
```

Comment:
This is pretty bad case of code duplication. Please factor out common patterns and refactor controller code if necessary.